lua/rustaceanvim/lsp/init.lua

```diff
@@ -249,11 +250,55 @@ M.reload_settings = function(bufnr)
   return clients
 end
 
+---Updates the target architecture setting for the LSP client associated with the given buffer.
+---@param bufnr? number The buffer number, defaults to the current buffer
+---@param target? string The target architecture. Defaults to the current buffer's target if not provided.
+M.set_target_arch = function(bufnr, target)
+  local function update_target(client)
+    -- Get the current target from the client's settings
+    local current_target = client.config.settings
+      and client.config.settings['rust-analyzer']
+      and client.config.settings['rust-analyzer'].cargo
+      and client.config.settings['rust-analyzer'].cargo.target
+
+    if not target then
+      if not current_target then
+        vim.notify('Using default OS target architecture.', vim.log.levels.INFO)
+      else
+        vim.notify('Target architecture is already set to the default OS target.', vim.log.levels.INFO)
+      end
+      return
+    end
+
+    if targets.target_is_valid_rustc_target(target) then
+      client.settings['rust-analyzer'].cargo.target = target
+      client.notify('workspace/didChangeConfiguration', { settings = client.config.settings })
+      vim.notify('Target architecture updated successfully to: ' .. target, vim.log.levels.INFO)
+      return
+    else
+      -- TODO: Log error when logging is implemented
```

Comment:
note: I'm probably not going to implement logging anytime soon. Let's remove these comments.
I'm not a fan of TODO comments anymore.